src/Storages/RocksDB/StorageEmbeddedRocksDB.cpp

```diff
@@ -448,15 +452,15 @@ void StorageEmbeddedRocksDB::initDB()
         }
     }
 
-    status = rocksdb::DB::Open(merged, rocksdb_dir, &db);
+    status = rocksdb::DBWithTTL::Open(merged, rocksdb_dir, &db, ttl, read_only);
```

Comment:
`DBWithTTL` inherits from `DB` so we can store one of them depending on the `ttl` value. I would prefer doing it like that, if `ttl` is `0`, create a regular `DB` otherwise create `DBWithTTL`.

Reply:
When `ttl` is non-positive, `DBWithTTL` behaves like the regular `DB`. It seems that there is no need to do that?

Reply:
I'm not sure about the overhead, all that extra ttl housekeeping doesn't seem like something we can just ignore.

You can define TTL as one of the engine arguments and create different DBs depending on that.